test/test_write.c

```diff
@@ -5,18 +5,67 @@ Written by Mark C. Miller, miller86@llnl.gov
 LLNL-CODE-707197. All rights reserved.
 
 This file is part of H5Z-ZFP. Please also read the BSD license
-https://raw.githubusercontent.com/LLNL/H5Z-ZFP/master/LICENSE 
+https://raw.githubusercontent.com/LLNL/H5Z-ZFP/master/LICENSE
 */
 
-#define _GNU_SOURCE
+#if !defined(_WIN32) && !defined(_GNU_SOURCE)
+#define _GNU_SOURCE /* ahead of ALL headers to take proper effect */
+#endif
 #include <assert.h>
 #include <errno.h>
 #include <fcntl.h>
-#include <math.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+
+#if defined(_WIN32)
+#define _USE_MATH_DEFINES
+#include <math.h>
+#define j0 _j0
+#include <io.h>
+#define strncasecmp _strnicmp
+#define strcasecmp _stricmp
+#define read _read
+#define open _open
+#define close _close
+#define srandom(X) srand(X)
+#define random rand
+
+// strcasestr is not available on Windows
+#include <ctype.h>
```

Comment:
See note above.

I guess we could move all this code which is used similiarly in the various tests harness `.c` files to a common place. Not suggestion *you* should do that...only asking for opinions about doing that.

Reply:
I think it would be a good idea to combine the common routines to one place.

Reply:
See PR #112